src/compiler/transformers/module/module.ts

```diff
@@ -1786,7 +1807,18 @@ namespace ts {
         scoped: true,
         text: `
             function __export(m) {
-                for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
+                for (var p in m) b(p);
+                function b(p) {
+                    if (!exports.hasOwnProperty(p))
+                        Object.create
+                            ? Object.defineProperty(exports, p, {

```

Comment:
Not a fan of how this gets checked on every iteration of the loop. In other places, we just decide up front what the helper is going to be and use that.